### DISCUSSION TITLE
How can I integrate Snort3 with my existing SO setup with Managersearch and Sensor node.

### DISCUSSION THREAD

**Discussion by kunalhiremath** [Security-Onion-Solutions/securityonion#9732] [Unsupported Versions]
Hi,
I want to integrate Snort3 to my existing Securityonion setup in which I have configured Managersearch and Sensor node. I want to integrate Snort IDS and forward the logs to elasticsearch. Is there any way to do so. Do you have any documentation for it.

**Comment by presianbg** [comment]
Hi,

Elastic has Snort integration by default:
https://docs.elastic.co/en/integrations/snort

In theory you should be able to:


<s>1. Install Elastic Agent in standalone mode and [configure it](https://www.elastic.co/guide/en/fleet/8.6/elastic-agent-configuration.html);
2. Send the snort logs to the Elastic Agent;
3. Ship the logs to the Security Onion ElasticSearch via the agent, do not forget to run the appropriate `so-allow` commands.

I haven't tried this path myself, but it should be working.</s>


EDIT: 
Or use the filebeat module:
https://docs.securityonion.net/en/2.3/filebeat.html#modules ,which should be way easier to incorporate. 

EDIT2:
Looks like few days ago there were similar discussion to this one:
https://github.com/Security-Onion-Solutions/securityonion/discussions/9705#discussioncomment-4851102

So... forget about what I said about Elastic Agent :) 

Cheers,
PY

**Reply by kunalhiremath** [reply]
Thanks @presianbg

**Reply by kunalhiremath** [reply]
Hey @presianbg  I followed your solution but I am facing some issues will you please check it out
I have posted the link below: 
https://github.com/Security-Onion-Solutions/securityonion/discussions/9886#discussion-4920911
